### PR TITLE
Fixed #26966 -- made MySQL backend skip defaults for JSON columns

### PR DESCRIPTION
**Pull request by adamchainz**
[Ticket](https://code.djangoproject.com/ticket/26966)

### CODE REVIEW

**Comment by claudep** [comment]
If `json` columns are not yet used by Django itself, I don't see the need for a release note. @timgraham ?

**Comment by timgraham** [comment]
Agreed.

File: django/db/backends/mysql/schema.py [hunk 1]
```diff
@@ -27,15 +27,16 @@ def quote_value(self, value):
 
     def skip_default(self, field):
         """
-        MySQL doesn't accept default values for TEXT and BLOB types, and
-        implicitly treats these columns as nullable.
+        MySQL doesn't accept default values for some data types, and implicitly
```
Comment (timgraham) [inline django/db/backends/mysql/schema.py]: Please chop the comma here.

Comment (adamchainz) [inline django/db/backends/mysql/schema.py]: Done

**Comment by adamchainz** [comment]
Release note removed.

**Comment by timgraham** [comment]
For future reference, our policy is not to use @ mentions in commit messages since GitHub spams those people with notifications.

**Comment by adamchainz** [comment]
Ok, my bad!

**Comment by avelis** [comment]
Is it appropriate to ask that this PR be backported to Django 1.9? Is it as simple as submitting a PR for it?

**Comment by adamchainz** [comment]
We don't normally backport things like this. You can "backport" yourself in your app by subclassing the schema editor with a copy of this method, then subclassing the database backend and reconfiguring your app to use that.

Django doesn't document the db backend API but it's fairly simple, and you can find examples [here](https://github.com/YPlan/django-perf-rec/tree/master/tests/django18_sqlite3_backend) and [here](https://github.com/carljm/django-transaction-hooks/tree/master/transaction_hooks/backends/mysql).

**Comment by timgraham** [comment]
For guidelines on what will be backported, see the [supported versions policy](https://docs.djangoproject.com/en/dev/internals/release-process/#supported-versions) -- 1.9 only receiving security fixes now and [will reach end-of-life](https://www.djangoproject.com/download/#supported-versions) in April.

**Pull request merged**